### ISSUE TITLE
Check Step Fails with 'method: HEAD'

### ISSUE THREAD

**Issue by 9numbernine9** [jgriff/http-resource#8]
Greetings!

I'm attempting to use the `http-resource` resource to check when a given URL has been updated. In my scenario, I'd like to just use an HTTP `HEAD` request in order to avoid downloading the remote URL each time (it's a large download and I only want to download it in my pipeline when it has actually changed).

I've created a minimal test pipeline that shows the error, using `www.wikipedia.org` as the target site:
```
resource_types:
- name: http-resource
  type: registry-image
  source:
    repository: jgriff/http-resource

resources:
- name: wikipedia
  type: http-resource
  check_every: 15m
  source:
    url: https://www.wikipedia.org
    version:
      header: last-modified
    method: HEAD

jobs:
- name: wikipedia-test
  plan:
  - get: wikipedia
    trigger: true

  - task: find
    config:
      image_resource:
        source:
          repository: starkandwayne/concourse
        type: registry-image
      platform: linux
      inputs:
      - name: wikipedia 
      run:
        path: /bin/bash
        args:
        - -c
        - |
          find .
```

Whenever the check step is executed, it simply enters an error state instead of checking the URL. The logs for the check don't really say what has gone wrong. :-( (Commenting out the `method: HEAD` part makes everything work correctly.) 
![image](https://user-images.githubusercontent.com/2762274/184138618-b626bbbb-5b55-400a-a99a-657ee32df4d0.png)

I should probably also mention that `wikipedia.org` does respond to `HEAD` requests and does set a `last-modiifed` header. :)
```
❯ curl -s -I https://www.wikipedia.org | grep -i 'last-modified'
last-modified: Wed, 27 Jul 2022 09:27:40 GMT
```

Any assistance you can offer would be appreciated!

**Comment by 9numbernine9** [comment]
Thanks for the analysis @jgriff ! FWIW I like the idea of special handling for translating `HEAD` into `--head` in order to preserve the abstraction over top of `curl` that this resource provides. :smile:

**Comment by jgriff** [comment]
This fix now available as of `latest` and [`0.4.0`](https://github.com/jgriff/http-resource/releases/tag/v0.4.0).

**Comment by 9numbernine9** [comment]
Works perfectly - thanks @jgriff ! :rocket:
